lib/calendarium-romanum/cli.rb

```diff
@@ -17,29 +17,37 @@ def query(date_str=nil)
       end
       sanctorale = data_file.load
 
-      date =
-        if date_str
-          begin
-            Date.parse(date_str)
-          rescue ArgumentError
-            die! 'Invalid date.'
+      if date_str
+        begin 
+          print_single_date(Date.parse(date_str), sanctorale)
+        rescue ArgumentError
+          if ((date_str =~ /((\d{4})-(\d{2})-(\d{2}))\z/) || (date_str =~ /((\d{4})\/(\d{2})\/(\d{2}))\z/)) # Year and month and day are given
```

Comment:
There is a lot of duplicate code here. Please, extract the date parsing logic to a new class (`CalendariumRomanum::Util` namespace is intended for internal tools like this).

Possible bonus points for handling all these cases in a single regular expression. :smile:

Reply:
No problem. Parser and even a single regular expression are already implemented. I will think about your second issue.